src/main/java/org/distributed/systems/chord/actors/Node.java

```diff
@@ -143,54 +148,97 @@ public Receive createReceive() {
-                    // Either Successor or Predecessor wants insert a new node.
-                    System.out.println("Node" + this.id + " is requested to confirm a JoinRequest");
-
-                    // Determine if a successor or predecessor
-                    if (msg.newPredecessor == null && msg.newSucessor != null) {
-                        // it's a successor
-                        this.sucessorId = msg.newSucessorKey;
-                        this.sucessor = msg.newSucessor;
-                        getContext().getSender().tell(new JoinMessage.JoinConfirmationReply(true), ActorRef.noSender());
-                        System.out.println("Node " + this.id + " confirmed the JoinRequest");
-                        System.out.println("Node's " + this.id + "current predecessor: " + this.predecessorId);
-                        System.out.println("Node's " + this.id + " new successor: " + msg.newSucessorKey);
-                    } else if (msg.newPredecessor != null && msg.newSucessor == null) {
-                        // it's a predecessor
-                        this.predecessorId = msg.newPredecessorKey;
-                        this.predecessor = msg.newPredecessor;
-                        JoinMessage.JoinConfirmationReply confirmReplyMsg = new JoinMessage.JoinConfirmationReply(true);
-                        getContext().getSender().tell(confirmReplyMsg, ActorRef.noSender());
-                        System.out.println("Node " + this.id + " confirmed the JoinRequest");
-                        System.out.println("Node's " + this.id + "current successor: " + this.sucessorId);
-                        System.out.println("Node's " + this.id + " new predecessor: " + msg.newPredecessorKey);
+                .match(FindSuccessor.Request.class, msg -> {
+                    // +1 to do inclusive interval
+                    // Single Node Edge Case: this.id == this.succId
+                    if (this.id == this.fingerTableSuccessor().id) {
+                        getContext().getSender().tell(new FindSuccessor.Reply(this.fingerTableSuccessor().chordRef, this.fingerTableSuccessor().id, msg.fingerTableIndex), getSelf());
+                    } else if (isBetweenExeclusive(this.id, this.fingerTableSuccessor().id + 1, msg.id)) {
+                        getContext().getSender().tell(new FindSuccessor.Reply(this.fingerTableSuccessor().chordRef, this.fingerTableSuccessor().id, msg.fingerTableIndex), getSelf());
                     } else {
-                        // Some illegal state, decline this request
-                        System.out.println("Node " + this.id + " DECLINED the JoinRequest");
-                        getContext().getSender().tell(new JoinMessage.JoinConfirmationReply(false), ActorRef.noSender());
+                        // this.fingerTableSuccessor().chordRef.forward(msg, getContext());
+                        ActorRef ndash = this.closest_preceding_node((int) msg.id);
```

Comment:
Shouldn't closest_preceding_node take a long?

Reply:
> Looks good! I think we can clean up a few things. Maybe make a separate class for fingerTable for example and clean up some logging.
> But I guess we can merge it into develop.

Okay I turned down verbosity a bit. Haven't moved out the entire fingertable stuff. Maybe we can move that into a service again.

Reply:
Don't know why the previous comment did get there, I though I was answering to your general feedback -> anyways, YES you are right. Will fix this